Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,708 @@
+define endian=big;
+define alignment=2;
+define space ram      type=ram_space      size=2 default;
+define space register type=register_space size=2;
+
+define register offset=0x00 size=2 [ R0 R1 R2 R3 R4 R5 R6 R7 ];
+define register offset=0x10 size=1 [ I D C O Z S ]; # Status bits
+define register offset=0x20 size=4 [ contextreg ];
+
+define token opcode_word (16)
+	clear           = (10, 15)
+	reg3_5          = (3, 5)
+	reg0_2          = (0, 2)
+	reg0_1          = (0, 1)
+	operation_size  = (2, 2)
+	swap_type       = (2, 2)
+	
+	branch_sign      = (5, 5)
+	branch_external  = (4, 4)
+	branch_condition = (0, 3)
+	
+	opcode6_9       = (6, 9)
+	opcode3_9       = (3, 9)
+	opcode2_9       = (2, 9)
+	opcode1_9       = (1, 9)
+	opcode0_9       = (0, 9)
+;
+
+define token jump_token (32)
+	unused_jump = (26, 31)
+	reg24_25    = (24, 25)
+	address_hi  = (18, 23)
+	jump_type   = (16, 17)
+	address_lo  = (0, 9)
+;
+
+define token double16 (32)
+	unused_lo = (17, 31)
+	value_lo = (16, 23)
+	unused_hi = (8, 15)
+	value_hi = (0, 7)
+	value = (0, 31)
+;
+
+define token immediate16 (16)
+	immvalue = (0, 15)
+	imm10    = (0, 9)
+;
+
+define context contextreg
+	doublebyte = (0, 0)
+;
+
+attach variables [ reg0_1        ] [ R0 R1 R2 R3             ];
+attach variables [ reg0_2 reg3_5 ] [ R0 R1 R2 R3 R4 R5 R6 R7 ];
+attach variables [ reg24_25      ] [ R4 R5 R6 R7             ]; # R7 is not used.
+
+################################################################
+
+jmpdest16:     reloc is address_hi & address_lo [ reloc = ((address_hi << 10) + address_lo) * 2; ] { export *:2 reloc;                         }
+branchdestb16: reloc is imm10                   [ reloc = inst_start + 4 - (imm10 * 2);          ] { export *:2 reloc;                         }
```

Comment:
I would suggest having 2 `branchdest16` using `imm10 & branch_sign=0` and `imm10 & branch_sign=1` then you would only need 1 branch instruction instead of 1 forwards and 1 backwards

Reply:
Right, I didn't know you could do that - this should be now sorted out.